sources/packages/web/src/components/common/applicationTracker/ApplicationInProgressChangeRequest.vue

```diff
@@ -0,0 +1,216 @@
+ */
+enum WaitingTypes {
+  MinistryApproval = "MinistryApproval",
+  ParentsDeclaration = "ParentsDeclaration",
+  PartnerDeclaration = "PartnerDeclaration",
+  StudentIncomeVerification = "StudentIncomeVerification",
+  ParentsIncomeVerification = "ParentsIncomeVerification",
+  PartnerIncomeVerification = "PartnerIncomeVerification",
+}
+
+export default defineComponent({
+  emits: {
+    changeRequestCancelled: () => true,
+  },
+  components: {
+    ApplicationStatusTrackerBanner,
+    ConfirmModal,
+  },
+  props: {
+    changeRequest: {
+      type: Object as PropType<ChangeRequestInProgressAPIOutDTO>,
+      required: false,
+    },
+    areApplicationActionsAllowed: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  setup(props, { emit }) {
+    const cancelChangeRequestModal = ref({} as ModalDialog<boolean>);
+    const snackBar = useSnackBar();
+    const cancelChangeRequestLoading = ref(false);
+
+    const viewChangeRequest = async () => {
+      if (!props.changeRequest) {
+        return;
+      }
+      const applicationId = props.changeRequest.applicationId;
+      const applicationProgramYear =
+        await ApplicationService.shared.getApplicationWithPY(applicationId);
+      await router.push({
+        name: StudentRoutesConst.DYNAMIC_FINANCIAL_CHANGE_REQUEST_APP_FORM,
+        params: {
+          selectedForm: applicationProgramYear.formName,
+          programYearId: applicationProgramYear.programYearId,
+          id: applicationId,
+        },
+      });
+    };
+
+    const cancelChangeRequest = async () => {
+      if (!props.changeRequest) {
+        return;
+      }
+      const confirm = await cancelChangeRequestModal.value.showModal();
+      if (!confirm) {
+        return;
+      }
+      try {
+        cancelChangeRequestLoading.value = true;
+        await ApplicationService.shared.applicationCancelChangeRequest(
+          props.changeRequest.applicationId,
+        );
+        emit("changeRequestCancelled");
+      } catch {
+        snackBar.error(
+          "There was an error cancelling your change request. Please try again.",
+        );
+      } finally {
+        cancelChangeRequestLoading.value = false;
+      }
+    };
+
+    /**
+     * Creates the list of processes currently waiting that are
+     * meaningful to be displayed to the student.
+     */
+    const waitingList = computed(() => {
+      if (!props.changeRequest) {
+        return [];
+      }
+      const waitingList: WaitingTypes[] = [];
+      const change = props.changeRequest;
+      if (
+        change.applicationEditStatus ===
+        ApplicationEditStatus.ChangePendingApproval
+      ) {
+        waitingList.push(WaitingTypes.MinistryApproval);
+      }
+      if (
+        [change.parent1Info, change.parent2Info].includes(
+          SuccessWaitingStatus.Waiting,
+        )
+      ) {
+        waitingList.push(WaitingTypes.ParentsDeclaration);
```

Comment:
What about pushing the messages to the array and populate the bullets based on array?

Reply:
> Opted for generating the list of waiting processes using v-if conditions to allow the plain texts to be expanded in the  > future and have elements other than plain text (e.g., links).